[EDL][OSD] Add setting to avoid commbreak notification

enen92 opened the pull request:
## Description
This PR adds an advanced setting to allow the user to avoid showing the edl commbreak notifications (they display by default)

```xml
<advancedsettings version="1.0">
        <edl>
		<displaycommbreaknotifications>false</displaycommbreaknotifications>
	</edl>
</advancedsettings>
```

## Motivation and context
Improve EDL usability

## How has this been tested?
Runtime tested

## What is the effect on users?
They may hide the notification that's shown when reaching EDL blocks.

## Screenshots (if appropriate):

![image](https://user-images.githubusercontent.com/7375276/147144158-a4e36428-e3f5-4b64-866a-7f4c79b7a473.png)


## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

lrusak reviewed (commented):
Code changes look fine 👍

enen92 commented:
Merging this as it has lukas "approval"

enen92 commented:
jenkins build and merge

ksooo commented on system/settings/settings.xml:
```diff
@@ -3724,6 +3724,13 @@
       </group>
+      <group id="2" label="39180">
+        <setting id="osd.showcommbreaknotification" type="boolean" label="39181" help="39182">
+          <level>2</level>
```
Is this edge case setting really level 2?

enen92 replied:
I am not sure either. I see advancedsettings.xml as something that should be avoided as much as possible, ultimately the place for changing settings that should not be changed unless the user really knows what he's doing (cache level, cache mode, etc). This setting avoids the display of a notification...Kodi is guaranteed to work well with or without this setting.
I see it as some gui setting at advanced or expert levels (more due to the fact it refers to EDL - which is not some main stream functionality - rather than what it actually does).
Level 2 is advanced, I can move it one level up if needed

ksooo commented on xbmc/cores/VideoPlayer/VideoPlayer.cpp:
```diff
@@ -2354,9 +2354,14 @@ void CVideoPlayer::CheckAutoSceneSkip()
-          StringUtils::SecondsToTimeString((edit.end - edit.start) / 1000, TIME_FORMAT_MM_SS);
-      CGUIDialogKaiToast::QueueNotification(g_localizeStrings.Get(25011), strTimeString);
+
+      std::shared_ptr<CSettings> settings = CServiceBroker::GetSettingsComponent()->GetSettings();
```
`const std::shared_ptr...`

ksooo commented:
Not sure this actually should be a GUI setting. I see this more as an advanced setting, if needed at all, because I consider this a real edge case. Just my 2ct, however.

enen92 commented:
There was one place missing where this notification was also triggered: if it started right at second 0 (demuxer initial position). Adjusted on the last revision and squashed.

lrusak commented:
jenkins build this please

enen92 commented:
Will move this to an advanced setting.

enen92 committed:
[EDL][OSD] Add setting to avoid commbreak notification

ksooo reviewed (approved):
LGTM

enen92 commented:
changed to `m_EdlDisplayCommbreakNotifications` for consistency on the last force-push (no functional changes)
